### ISSUE TITLE
different lang with different rules

### ISSUE THREAD

**Issue by Airkro** [vuejs/eslint-plugin-vue#1523]
**Tell us about your environment**

- **ESLint version:** 7.29
- **eslint-plugin-vue version:** 7.12.0
- **Node version:** 14.17.0

**The problem you want to solve.**

```vue
<!-- use different rule -->
<script lang="js"></script>
<script lang="ts"></script>
```

**Your take on the correct solution to problem.**

```js
// .eslintrc.js
module.export = {
  parser: 'vue-eslint-parser',
  overrides: [
    {
      files: '*.vue/**/*.ts',
      parserOptions: {
        parser: '@typescript-eslint/parser',
      },
    },
    {
      files: '*.vue/**/*.js',
      parserOptions: {
        parser: '@babel/eslint-parser',
      },
      rule: {
        'no-unused-vars': 'warn',
      }
    }
  ]
};
```

**Additional context**

Something like:
https://github.com/eslint/eslint-plugin-markdown#advanced-configuration

**Comment by ota-meshi** [comment]
Thank you for posting this issue.
But, this plugin cannot be configured separately for `lang` due to its mechanism.
Thank you for your understanding.

**Comment by Airkro** [comment]
https://github.com/vuejs/vue-eslint-parser/pull/116

It's possible, right?

**Comment by adamxi** [comment]
@Airkro: I think I got it all working doing this:

```
	parser: "vue-eslint-parser",
	parserOptions: {
		parser: {
			// Script parser for `<script>`
			js: false,

			// Script parser for `<script lang="ts">`
			ts: "@typescript-eslint/parser",

			// Script parser for vue directives (e.g. `v-if=` or `:attribute=`) and vue interpolations (e.g. `{{variable}}`).
			// If not specified, the parser determined by `<script lang ="...">` is used.
			"<template>": false,
		},
	},
```

I also had to an install of the typescript linter:
`npm i --save-dev typescript @typescript-eslint/parser`

**Comment by ota-meshi** [comment]
Rules cannot be configured by lang.

**Comment by lovetingyuan** [comment]
> Rules cannot be configured by lang.

Why not use 'processor.preprocess' ?
